### PR TITLE
ACU Omnibus, July 2023

### PR DESCRIPTION
**Pull request by mhasself**
## Description

Bunch of bugfixes and features additions for ACU.

- The scan speed and acceleration used by `generate_scan` can be configured through the new `set_scan_params` task; the starting values are set, on startup, based on the platform type (configured in agent code preamble).
- In `generate_scan`, make more arguments optional; only `az_endpoint*` are required.
- Reduce data rates from the `monitor` process, for both aggregator and to influx feeds.  This is especially helpful for influx (for aggregator, UDP feed dominates total volume).
- Fix `generate_scan` so that it doesn't ever upload fewer than 4 points on the start of a leg; this left it vulnerable to causing an ACU fault if the scan was terminated after such a partial upload.
- Other small changes and fixes.

## How Has This Been Tested?

These have been tested as they evolved on SATP1 at site.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: socs/agents/acu/agent.py [hunk 1]
```diff
@@ -401,18 +436,20 @@ def monitor(self, session, params):
...
-                            if key in self.monitor_fields[category]:
-                                if isinstance(value, bool):
-                                    self.data['status'][category][self.monitor_fields[category][key]] = int(value)
-                                elif isinstance(value, int) or isinstance(value, float):
-                                    self.data['status'][category][self.monitor_fields[category][key]] = value
-                                elif value is None:
-                                    self.data['status'][category][self.monitor_fields[category][key]] = float('nan')
-                                else:
-                                    self.data['status'][category][self.monitor_fields[category][key]] = str(value)
+                if k in not_data_keys:
```
Comment (BrianJKoopman) [inline socs/agents/acu/agent.py:439]: Worth still checking if `instance(v, dict)`? The current way this is seems to assume all non-dicts are added to `not_data_keys`.

File: socs/agents/acu/agent.py [hunk 2]
```diff
@@ -932,6 +971,53 @@ def get_history(t):
             msg = 'Irregularity during motion!'
         return success, msg
 
+    @inlineCallbacks
+    def _go_to_axes(self, session, el=None, az=None, third=None):
+        """Execute a movement along multiple axes, using "Preset" mode.  This
+        just launches _got_to_axis on each required axis, and collects
```
Comment (BrianJKoopman) [inline socs/agents/acu/agent.py:977]: `_got_to_axis` -> `_go_to_axis`

File: socs/agents/acu/agent.py [hunk 3]
```diff
@@ -1035,6 +1104,38 @@ def set_boresight(self, session, params):
...
+        # individual items rather than creating a new dict here.
+        for k, v in DEFAULT_SCAN_PARAMS[self.acu_config['platform']].items():
+            self.scan_params[k] = v
+
+    @ocs_agent.param('az_speed', type=float, default=None)
+    @ocs_agent.param('az_accel', type=float, default=None)
+    @ocs_agent.param('reset', default=False, type=bool)
+    @inlineCallbacks
+    def set_scan_params(self, session, params):
+        """set_scan_params()
```
Comment (BrianJKoopman) [inline socs/agents/acu/agent.py:1118]: `set_scan_params()` -> `set_scan_params(az_speed=None, az_accel=None, reset=False)`

File: socs/agents/acu/agent.py [hunk 4]
```diff
@@ -1035,6 +1104,38 @@ def set_boresight(self, session, params):
...
+
+        **Task** - Update the default scan parameters, used by
+        generate_scan if not passed explicitly.
+
+        Parameters:
+          az_speed (float): The azimuth scan speed.
+          az_accel (float): The (average) azimuth acceleration at turn-around.
+          reset (bool): If True, reset all params to default values
```
Comment (BrianJKoopman) [inline socs/agents/acu/agent.py:1126]: These all seem optional, so I would have `(type, optional)` here.

File: socs/agents/acu/agent.py [hunk 5]
```diff
@@ -1173,8 +1274,8 @@ def line_batcher(lines, n=10):
 
     @ocs_agent.param('az_endpoint1', type=float)
     @ocs_agent.param('az_endpoint2', type=float)
-    @ocs_agent.param('az_speed', type=float)
-    @ocs_agent.param('az_accel', type=float)
+    @ocs_agent.param('az_speed', type=float, default=None)
+    @ocs_agent.param('az_accel', type=float, default=None)
```
Comment (BrianJKoopman) [inline socs/agents/acu/agent.py:1278]: Update the method signature override to have `az_speed=None` and `az_accel=None`.

**Review by BrianJKoopman — CHANGES_REQUESTED** [review]
This looks good to me. A few small docs changes and one question.

EDIT: Also, can you merge in the latest from `main` to fix docs builds?

File: socs/agents/acu/agent.py (hunk 1 above)
Comment (mhasself) [reply socs/agents/acu/agent.py:439]: I think it's simpler to enforce the use of not_data_keys for all non-data keys.  If that list is kept up to date, no non-dicts will get through to the loop over .items().

**Comment by mhasself** [comment]
> This looks good to me. A few small docs changes and one question.
> 
> EDIT: Also, can you merge in the latest from `main` to fix docs builds?

Done and done -- thanks!

**Review by BrianJKoopman — APPROVED** [review]
Awesome, thanks for the updates!

**Pull request merged**